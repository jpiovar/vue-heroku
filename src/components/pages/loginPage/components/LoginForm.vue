<template>
<div class="login-form">
  <form>
    <div class="form-group">
      <label for="userName">User name</label>
      <input type="text" v-model="userNameVal" class="form-control" id="userName" aria-describedby="userNameHelp" placeholder="Enter user name">
      <!-- <small id="userNameHelp" class="form-text text-muted">Please use valid user name.</small> -->
    </div>
    <div class="form-group">
      <label for="exampleInputPassword1">Password</label>
      <input type="password" v-model="passwordVal" class="form-control" id="exampleInputPassword1" placeholder="Password">
    </div>
    <button type="submit" class="btn btn-primary" @click.prevent="loginForm">Login</button>
  </form>
  <div class="alert alert-warning alert-dismissible fade show" role="alert" v-if='showAlert'>
    Please use valid credentials.
    <button type="button" class="close" aria-label="Close" @click="hideAlertBanner">
      <span aria-hidden="true">&times;</span>
    </button>
  </div>
</div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

@Component({
  components: {},
  props: {
    loginRef: {
      required: true,
      type: Function
    }
  }
})
export default class LoginForm extends Vue {
  loginRef!: any;
  userNameVal: string = '';
  passwordVal: string = '';
  showAlert: boolean = false;

  loginForm() {
    this.loginRef({'name': this.userNameVal, 'password': this.passwordVal});
  }

  showAlertBannerAndReset() {
    this.showAlert = true;
    this.userNameVal = '';
    this.passwordVal = '';
  }
  hideAlertBanner() {
    this.showAlert = false;
  }
}
</script>

<style lang="scss" scoped>
.login-form {
    height: 400px;
    text-align: left;
    form {
      margin: 20px 0;
    }
}
</style>
