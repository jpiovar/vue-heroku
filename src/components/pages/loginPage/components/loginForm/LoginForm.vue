<template>
<div class="login-form">
  <div class="card shadow p-3 mb-5 bg-white rounded">
    <div class="card-body">
      <h4 class="card-title text-center">Simple Quora Login</h4>
      <form @submit.prevent="loginForm">
        <div class="form-group">
          <label for="userName">User name</label>
          <input type="text"
            v-model="userNameVal"
            ref="userName" class="form-control" id="userName"
            aria-describedby="userNameHelp" placeholder="Type user name">
        </div>
        <div class="form-group">
          <label for="exampleInputPassword1">Password</label>
          <input type="password"
            v-model="passwordVal"
            class="form-control" id="exampleInputPassword1" placeholder="Type password">
        </div>
        <button type="submit" class="btn btn-primary"
          @keydown.tab.prevent="setFocus">Login
        </button>
      </form>
      <div class="alert alert-warning alert-dismissible fade show" role="alert" v-if='showAlert'>
        Please use valid credentials.<br/> janko/janko or hrasko/hrasko 
        <button type="button" class="close" ref="alertBtn" aria-label="Close"
          @click.prevent="hideAlertBanner"
          @keydown.tab.prevent="setFocusInput">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
    </div>
  </div>
</div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

@Component({
  components: {},
  props: {
    loginRef: {
      required: true,
      type: Function,
    },
  },
})
export default class LoginForm extends Vue {
  loginRef!: any;

  userNameVal: string = '';

  passwordVal: string = '';

  showAlert: boolean = false;

  loginForm() {
    this.loginRef({ name: this.userNameVal, password: this.passwordVal });
    // this.userNameValGS = `name ${this.userNameValGS}` ;
  }

  showAlertBannerAndReset() {
    this.showAlert = true;
    this.userNameVal = '';
    this.passwordVal = '';
  }

  hideAlertBanner() {
    this.showAlert = false;
    (this.$refs.userName as HTMLFormElement).focus();
  }

  setFocus() {
    if (!this.showAlert) {
      (this.$refs.userName as HTMLFormElement).focus();
    } else if (this.showAlert && this.$refs.alertBtn) {
      (this.$refs.alertBtn as HTMLFormElement).focus();
    }
  }

  setFocusInput() {
    (this.$refs.userName as HTMLFormElement).focus();
  }

  get userNameValGS(): string {
    return this.userNameVal;
  }

  set userNameValGS(val: string) {
    this.userNameVal = val;
  }
}
</script>

<style lang="scss" scoped>
.login-form {
    text-align: left;
    form {
      margin: 20px 0;
    }
}
</style>
