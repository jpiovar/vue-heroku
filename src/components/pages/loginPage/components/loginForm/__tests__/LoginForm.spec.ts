import { shallowMount, mount, Wrapper } from '@vue/test-utils';
import LoginForm from '@/components/pages/loginPage/components/loginForm/LoginForm.vue';
import Vue from 'vue';

// const Constructor = Vue.extend(LoginForm);
// const vm = new Constructor().$mount();

const props = {
    loginRef() { },
};

const getWrapperShallow = (propsData = props): Wrapper<LoginForm> => shallowMount(LoginForm, {
    propsData: {
        ...propsData
    },
});

const getWrapperMount = (propsData = props): Wrapper<LoginForm> => mount(LoginForm, {
    propsData: {
        ...propsData
    },
});

describe('LoginForm.vue', () => {
    it('should match the snapshot', () => {
        const wrapper: Wrapper<LoginForm> = getWrapperShallow();
        expect(wrapper.vm.$el).toMatchSnapshot()
    });
    it('renders input id userName', () => {
        const wrapper: Wrapper<LoginForm> = getWrapperShallow();
        const inputField = wrapper.find('input#userName');
        expect(inputField.exists()).toBe(true);
        expect(inputField.isVisible()).toBe(true);
    });
    it('renders input id exampleInputPassword1', () => {
        const wrapper: Wrapper<LoginForm> = getWrapperShallow();
        const inputField = wrapper.find('input#exampleInputPassword1');
        expect(inputField.exists()).toBe(true);
        expect(inputField.isVisible()).toBe(true);
    });
    it('calls method loginForm after form submit done', () => {
        const wrapper: any = getWrapperShallow();
        spyOn(wrapper.vm, 'loginForm');
        wrapper.find("form").trigger("submit.prevent");
        expect(wrapper.vm.loginForm).toHaveBeenCalled();
    });
});