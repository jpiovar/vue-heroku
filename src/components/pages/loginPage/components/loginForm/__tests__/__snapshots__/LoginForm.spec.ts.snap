// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm.vue should match the snapshot 1`] = `
<div
  class="login-form"
>
  <div
    class="card shadow p-3 mb-5 bg-white rounded"
  >
    <div
      class="card-body"
    >
      <h4
        class="card-title text-center"
      >
        Simple Quora Login
      </h4>
       
      <form>
        <div
          class="form-group"
        >
          <label
            for="userName"
          >
            User name
          </label>
           
          <input
            aria-describedby="userNameHelp"
            class="form-control"
            id="userName"
            placeholder="Type user name"
            type="text"
          />
        </div>
         
        <div
          class="form-group"
        >
          <label
            for="exampleInputPassword1"
          >
            Password
          </label>
           
          <input
            class="form-control"
            id="exampleInputPassword1"
            placeholder="Type password"
            type="password"
          />
        </div>
         
        <button
          class="btn btn-primary"
          type="submit"
        >
          Login
        
        </button>
      </form>
       
      <!---->
    </div>
  </div>
</div>
`;
